File: backend/src/main/java/com/example/model/ClozeNote.java
package com.example.model;


import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.Embeddable;
import javax.persistence.Entity;

@Embeddable
public class ClozeNote{
    @Getter
    @Setter
    String text;

    ClozeNote(){}

    public ClozeNote(String text){
        this.text = text;
    }

    @Override
    public String toString() {
        return "Cloze Note {text:" + text + "}";
    }
}

File: src/types/ClozeNote.ts
import Note from 'Note';

export default class ClozeNote extends Note {

    private _text: string;

    constructor(text: string, id: string) {
        super(id);
        this._text=text;
    }

    get text(): string {
        return this._text;
    }

    set text(text: string) {
        this._text = text;
    }
}